### PR TITLE
Common: Try to improve MSP doc especially for HDZero

### PR DESCRIPTION
**Pull request by timtuxworth**
While setting up my HDZero goggles, I found some of the wiki relating to MSP could be clearer. This is my attempt to improve it.

I've also re-structure the page, there was a lot of general stuff relating to OSD comments that was buried under the DJI Goggles section, so I've pushed the different google setups to the top and then made a common section below for all the stuff that applies regardless which goggles you have.

Also clarified how you use the new Mission Planner "HD Layout" OSD feature.

### CODE REVIEW

File: common/source/docs/common-msp-osd-overview-4.2.rst
Comment (Hwurzburg) [inline common/source/docs/common-msp-osd-overview-4.2.rst:49]: this is incorrect....this is push mode for MSP OSDs using only the TX line not dji goggles

File: common/source/docs/common-msp-osd-overview-4.2.rst
Comment (Hwurzburg) [inline common/source/docs/common-msp-osd-overview-4.2.rst:55]: Moving all of this before the panel items section is wrong...these are MSP only panel items...DisplayPort and integrated OSDs display MANY more panels (all of them that are not MSP only)....

**Review by Hwurzburg — CHANGES_REQUESTED** [review]
just quick first pass...will take some more time this weekend

File: common/source/docs/common-msp-osd-overview-4.2.rst
Comment (timtuxworth) [reply common/source/docs/common-msp-osd-overview-4.2.rst:55]: Hmm -It is VERY confusing to have all these OSD values listed on the page and have to scroll past all this stuff to get to the DisplayPort part. A couple of options, what do you think:
1. Put DisplayPort first - then the OSD list can be at the bottom of the page and only those with a telemetry OSD have to scroll there to see it
2. Split this into two pages.

File: common/source/docs/common-msp-osd-overview-4.2.rst
Comment (timtuxworth) [reply common/source/docs/common-msp-osd-overview-4.2.rst:49]: Ok - I have to say I don't really understand the MSP_OPTIONS bit 1 then, all I know is that if I set it to 1 my HDZero goggles don't work. So when would you ever set it to 1?

Comment (Hwurzburg) [reply common/source/docs/common-msp-osd-overview-4.2.rst:49]: on an msp osd that is connected only by its rx input to the uarts tx output...normally the osd requests that info be semt..this option allows it to be 'pushed' to the osd

File: common/source/docs/common-msp-osd-overview-4.2.rst
Comment (timtuxworth) [reply common/source/docs/common-msp-osd-overview-4.2.rst:55]: Thanks @Hwurzburg - I'll try that.

File: common/source/docs/common-msp-osd-overview-4.2.rst
Comment (timtuxworth) [reply common/source/docs/common-msp-osd-overview-4.2.rst:49]: So the bit itself "EnableTelemetryMode" - should actually be "EnableTelemetryPollingMode" ?

Comment (timtuxworth) [reply common/source/docs/common-msp-osd-overview-4.2.rst]: > this is incorrect....this is push mode for MSP OSDs using only the TX line not dji goggles

So which goggles would this apply to?

Comment (Hwurzburg) [reply common/source/docs/common-msp-osd-overview-4.2.rst]: a standalone osd like mwosd

Comment (timtuxworth) [reply common/source/docs/common-msp-osd-overview-4.2.rst]: So if I say this: 
.. note:: a standalone osd like MWOSD, with only one wire connected, will need :ref:`MSP_OPTIONS<MSP_OPTIONS>` bit 0 = 1 (EnableTelemetryMode). This is the only case where this should be used.

Comment (Hwurzburg) [reply common/source/docs/common-msp-osd-overview-4.2.rst]: I am working on a rewrite to make the various osd types clearer..standby

**Comment by timtuxworth** [comment]
Made some major changes based on your feedback. I'm thinking it flows a lot better now - try reading it "from the top" and see if it makes sense and if I've broken anything.

**Comment by Hwurzburg** [comment]
added info on non dji msp....hopefully this is clear now

**Comment by timtuxworth** [comment]
> added info on non dji msp....hopefully this is clear now

This is definitely an improvement. I think overall much clearer and easier to understand and also easier to find what you need for your specific goggles. I'll do a QA pass probably this afternoon.

File: common/source/docs/common-msp-osd-overview-4.2.rst
Comment (timtuxworth) [inline common/source/docs/common-msp-osd-overview-4.2.rst:30]: I disagree strongly with you about embedding this table here @Hwurzburg . Sure it makes sense if you have this type of goggles, but for anyone else it makes the page very hard to read and find what you are looking for. It's your call of course.

File: common/source/docs/common-msp-osd-overview-4.2.rst
Comment (timtuxworth) [inline common/source/docs/common-msp-osd-overview-4.2.rst:136]: This discussion of fonts now reads like it only applies to DisplayPort. I'm going to suggest a change which moves it down into the OSD section.

File: common/source/docs/common-msp-osd-overview-4.2.rst
Comment (Hwurzburg) [reply common/source/docs/common-msp-osd-overview-4.2.rst:208]: TXT _RES only applies to DisplayPort systems with their own fonts and display resolutions...its merely a value shipped to the Display port device

**Comment by timtuxworth** [comment]
The new version reads like the ability to have two OSDs (integrated + Air Unit) does not apply to DisplayPort. As written you can enable Telemetry based OSD + internal or DJI Custom OSD + internal, but NOT DisplayPort + internal.
Is that correct @Hwurzburg ?

File: common/source/docs/common-msp-osd-overview-4.2.rst
Comment (Hwurzburg) [reply common/source/docs/common-msp-osd-overview-4.2.rst:37]: it belongs with the MSP only stuff

**Comment by Hwurzburg** [comment]
not unless you use the second osd type feature added in 4.5
I must say that the last changes you made (aside from some minor format corrections) I find unacceptable...and I dont have time to individually work on them any more right now....so we will just put this on hold for now

**Comment by timtuxworth** [comment]
> not unless you use the second osd type feature added in 4.5
> I must say that the last changes you made (aside from some minor format corrections) I find unacceptable...and I dont have time to individually work on them any more right now....so we will just put this on hold for now

Totally get it Henry - you have a lot on, I don't know how you do it. I'm also working on a video just around ArduPilot with HDZero, so no rush.

**Comment by Hwurzburg** [comment]
I have included a lot of your changes from the previous push that  I commented on...I am merging this without the later pushes..please open a new PR if you want to make additional changes to make review easier..

**Comment by Hwurzburg** [comment]
@timtuxworth I had a git meltdown and lost all of my changes!! going to take a break and restart later...sorry for the FU

**Comment by timtuxworth** [comment]
Thanks for working with me on this @Hwurzburg . I'm pretty sure we've made this a lot easier to follow and to understand how to set up any specific set of goggles. I'm going to do some flying next! :-)

**Pull request merged**
